wp-content/civi-extensions/goonjcustom/api/v3/Goonjcustom/Cron.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+/**
+ * Civirules.Cron API specification (optional)
+ * This is used for documentation and validation.
+ *
+ * @param array $spec description of fields supported by this API call
+ *
+ * @return void
+ */
+function _civicrm_api3_goonjcustom_cron_spec(&$spec) {
+  //there are no parameters for the civirules cron
+}
+
+/**
+ * Civirules.Cron API
+ *
+ * @param array $params
+ *
+ * @return array API result descriptor
+ * @see civicrm_api3_create_success
+ * @see civicrm_api3_create_error
+ * @throws \CRM_Core_Exception
+ */
+function civicrm_api3_goonjcustom_cron($params) {
+  $returnValues = [];
+
+  try {
+
+    $currentDate = date('Y-m-d');
+
+    $activity_assignees = \Civi\Api4\Activity::get(TRUE)
+        ->addSelect('target_contact_id', 'activity_contact.contact_id', 'activity_date_time')
+        ->addJoin('ActivityContact AS activity_contact', 'LEFT', ['activity_contact.record_type_id', '=', 1])
+        ->addWhere('activity_type_id', '=', 57)
+        ->addWhere('status_id', '=', 1)
+        ->addWhere('activity_date_time', '>=', $currentDate . ' 00:00:00')
+        ->addWhere('activity_date_time', '<=', $currentDate . ' 23:59:59')
```

Comment:
Can we define two variables like this:
```php
// Start of the day
$startOfDay = new DateTime('today midnight');

// End of the day
$endOfDay = new DateTime('tomorrow midnight -1 second');
```

and then do:

```suggestion
        ->addWhere('activity_date_time', '>=', $startOfDay->format('Y-m-d H:i:s'))
        ->addWhere('activity_date_time', '<=', $endOfDay->format('Y-m-d H:i:s'))
```

Reply:
sure updated